core/src/wsv.rs

```diff
@@ -83,14 +83,13 @@ impl World {
 }
 
 /// Current state of the blockchain aligned with `Iroha` module.
-#[derive(Debug)]
 pub struct WorldStateView {
     /// The world - contains `domains`, `triggers`, etc..
     pub world: World,
     /// Configuration of World State View.
     pub config: Configuration,
     /// Blockchain.
-    blocks: Arc<Chain>,
+    pub block_hashes: std::cell::RefCell<Vec<HashOf<VersionedCommittedBlock>>>,
```

Comment:
Because the hashes are inexpensive to store. If we go to kura we are engaging in cross thread communication. Also in the case of rollbacks there can be a situation where kura is aware of more blocks then the world state is.